api/src/main/java/io/opentelemetry/api/metrics/batch/BatchObserverContext.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.metrics.batch;
+
+import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.api.metrics.Instrument;
+import io.opentelemetry.api.metrics.batch.AsynchronousBatchInstrument.Observation;
+import java.util.function.Consumer;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Util class for creating multiple instruments that can be atomically observed in a single
+ * callback.
+ *
+ * <p>This class is equivalent to individually calling {@code observe} on every {@link Instrument},
+ * but has the advantage of all recordings occurring atomically.
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * meter.newBatchObserver(
+ *     context -> {
+ *       DoubleSumBatchObserver timeObserver =
+ *           context.doubleSumObserverBuilder("time").setUnit("s").build();
+ *       LongSumBatchObserver distanceObserver =
+ *           context.longSumObserverBuilder("distance").setUnit("m").build();
+ *
+ *       context.registerCallback(
+ *           result ->
+ *               MyMeasure measure = MyMeasure.getdata();
+ *               result.observe(
+ *                   Labels.of("myKey", "myValue"),
+ *                   distanceObserver.observation(measure.getDistance()),
+ *                   timeObserver.observation(measure.getTime())));
+ *     });
+ * }</pre>
+ */
+@ThreadSafe
+public interface BatchObserverContext extends Instrument {
+
+  /**
+   * Returns a new builder for a {@link DoubleValueBatchObserver}.
+   *
+   * @param name the name of the instrument. Should be a ASCII string with a length no greater than
+   *     255 characters.
+   * @return a new builder for a {@code DoubleValueBatchObserver}.
+   * @throws NullPointerException if {@code name} is null.
+   * @throws IllegalArgumentException if different metric with the same name already registered.
+   * @throws IllegalArgumentException if the {@code name} does not match the requirements.
+   */
+  DoubleValueBatchObserver.Builder doubleValueObserverBuilder(String name);
```

Comment:
AFAIR, we agreed to do not double the instruments before a better specification for the batch observer is done. I cannot find a link that documents this decision, so maybe it was just a conversation between me and @jkwatson

Reply:
Would be good to have that clarified.

The reason I went with separate instruments was to prevent the ability for developers to call `setCallback()` on the `Instrument` as it could lead to confusion.

Reply:
I'm on both sides of the fence. I'd rather not duplicate the instruments, but I'd also like to make the API really clear. Is there a way we can do both? Suggestions welcome!

Reply:
Could we use the same instrument, but have the batch one have a no-op for the callback addition? Or could we make the non-batch one extend the interface for the batch one and add the callback method?

Reply:
I'd also welcome suggestions as I couldn't come up with a "clean" way that didn't result in tradeoffs in one place or another.

We could use the same instrument and make `setCallback()` a no-op. My concern would be developers use it and don't realize it's a problem until they wonder why the metric isn't being observed. Maybe there's a way to mitigate that in the actual SDK implementation.

We could also have a non-batch extend a batch interface. A possible concern is that `observation()` is then available on the non-batch Instrument. As with the previous no-op suggestion, maybe there's a way to handle that gracefully.

At the moment there are options, I'm not certain which way we should jump

Reply:
ok, here are my thoughts:
1) If I have to choose between making things clear and easy for end users and making our lives easier as maintainers, I'll choose the end user. This is an argument in favor of duplicating the instruments.
2) If we like this pattern, we could use it for the non-batch observer instruments as well. In that case, could we convert both types over to the new ones and get rid of the old ones?

Reply:
Wholeheartedly agree with 1).

2) I'm a little unsure of exactly what you mean. Can you clarify what the "pattern" is that you refer to?

It might be possible to use a single instrument for both cases, would need to experiment a bit. Especially as there is a difference between `observe(value, labels)` and `observation(value)`.

Reply:
@kenfinnigan check out https://github.com/open-telemetry/opentelemetry-java/pull/2120 where I've deprecated the `setCallback` method on the Async instruments. I think this might mean we could re-use the same instruments, yes?

Reply:
I think that PR allows for the reuse of the instruments, yes!

Reply:
Will take a look, thanks @jkwatson

Reply:
@jkwatson I haven't moved things around, but I think you're right that removing `setCallback` from `AsynchronousInstrument` makes it possible to re-use the existing instrument builders.

Would that also mean `observation()` would be added to the builder?

Reply:
not sure how that would work, since it needs to return an `Observation`. Maybe I'm missing what you mean by the question.

Reply:
@jkwatson True, was trying to think it through without spending time reworking all the code. ;-)

I think I was getting confused, so ignore that comment for now.

What's the outlook on #2120 being merged?

Reply:
Seems very good. I'm letting it rest a bit, since it's an API change, and it's a holiday week. I would assume at this point that it will happen, unless someone comes up with a very good reason that it shouldn't.